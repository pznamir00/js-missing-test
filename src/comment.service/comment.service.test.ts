import { GitHub } from '@actions/github/lib/utils'
import createComment from './comment.service'

describe('comment service', () => {
  describe('createComment', () => {
    it('calls rest.issues.createComment', async () => {
      const { octokit, owner, repo, prNumber } = setup()
      await createComment(octokit, owner, repo, prNumber, [
        {
          srcPath: 'src/file1.js',
          testPath: 'src/file1.test.js'
        },
        {
          srcPath: 'src/file2.js',
          testPath: 'src/file2.test.js'
        },
        {
          srcPath: 'src/file3.js',
          testPath: 'src/file3.test.js'
        },
        {
          srcPath: 'src/file4.js',
          testPath: 'src/file4.test.js'
        }
      ])
      expect(
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        octokit.rest.issues.createComment.mock.calls[0][0].body
      ).toMatchSnapshot()
    })
  })
})

function setup() {
  const octokit = {
    rest: {
      issues: {
        createComment: jest.fn()
      }
    }
  } as unknown as InstanceType<typeof GitHub>
  const owner = 'owner123'
  const repo = 'test-project-123'
  const prNumber = 2137
  return { octokit, owner, repo, prNumber }
}
