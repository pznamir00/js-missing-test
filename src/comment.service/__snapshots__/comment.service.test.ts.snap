// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comment service createComment calls rest.issues.createComment 1`] = `
"
    Pull request #2137 is missing 4 tests:
    - source file: src/file1.js is missing src/file1.test.js test file
- source file: src/file2.js is missing src/file2.test.js test file
- source file: src/file3.js is missing src/file3.test.js test file
- source file: src/file4.js is missing src/file4.test.js test file

    "
`;
